src/traces/choropleth/hover.js

```diff
@@ -12,23 +12,35 @@
 var Axes = require('../../plots/cartesian/axes');
 var attributes = require('./attributes');
 
-module.exports = function hoverPoints(pointData) {
+module.exports = function hoverPoints(pointData, xval, yval) {
     var cd = pointData.cd;
     var trace = cd[0].trace;
     var geo = pointData.subplot;
 
-    // set on choropleth paths 'mouseover'
-    var pt = geo.choroplethHoverPt;
+    var pt, i, j, isInside;
 
-    if(!pt) return;
+    for(i = 0; i < cd.length; i++) {
+        pt = cd[i];
+        isInside = false;
 
-    var centroid = geo.projection(pt.properties.ct);
+        if(pt._polygons) {
+            for(j = 0; j < pt._polygons.length; j++) {
+                if(pt._polygons[j].contains([xval, yval])) {
+                    isInside = !isInside;
```

Comment:
Just to have a record of it - @etpinard and I discussed this in chat: in principle it would be faster to keep the nesting structure of the geojson, that tells us the difference between exterior and internal polygons. Then we could:
- Look for the first exterior loop we are inside
- Don’t check any more exterior loops after that
- Only check the interior loops for that particular exterior loop

But for now there can only be at most 200 locations, for ~ 200 countries, and polygon.contains should short-circuit most of these based on their bounding boxes, so this is not a big deal. If we ever add US counties or something we can revisit. But then @etpinard points out that this becomes complicated in a few obnoxious cases of [higher-order nesting](https://en.wikipedia.org/wiki/India%E2%80%93Bangladesh_enclaves) which this simple but slower algorithm (are you inside an odd number of polygons?) gets right.